### ISSUE TITLE
CI: K8sServicesTest Checks service across nodes TFTP with DNS Proxy port collision Tests TFTP from DNS Proxy Port

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#12336]
## CI failure

```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/1.15-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Request from testclient-t6n6h pod to service tftp://192.168.36.12:30696/hello --local-port 39623 failed
Expected command: kubectl exec -n default testclient-t6n6h -- /bin/bash -c 'fails=""; id=$RANDOM; for i in $(seq 1 10); do if curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 20 tftp://192.168.36.12:30696/hello --local-port 39623 -H "User-Agent: cilium-test-$id/$i"; then echo "Test round $id/$i exit code: $?"; else fails=$fails:$id/$i=$?; fi; done; if [ -n "$fails" ]; then echo "failed: $fails"; fi; cnt="${fails//[^:]}"; if [ ${#cnt} -gt 0 ]; then exit 42; fi' 
To succeed, but it failed:
Exitcode: 42 
Err: Process exited with status 42
Stdout:
 	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/1 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/2 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/3 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/4 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/5 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/6 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/7 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/8 exit code: 0
	 
	 Hostname: testds-v6vbk
	 
	 Request Information:
	 	client_address=10.0.0.231
	 	client_port=39623
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 589/9 exit code: 0
	 failed: :589/10=72
	 
Stderr:
 	 command terminated with exit code 42
	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/1.15-gopath/src/github.com/cilium/cilium/test/k8sT/Services.go:1178
```

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/3247/testReport/junit/Suite-k8s-1/15/K8sServicesTest_Checks_service_across_nodes_TFTP_with_DNS_Proxy_port_collision_Tests_TFTP_from_DNS_Proxy_Port/

[e1aaccbf_K8sServicesTest_Checks_service_across_nodes_TFTP_with_DNS_Proxy_port_collision_Tests_TFTP_from_DNS_Proxy_Port.zip](https://github.com/cilium/cilium/files/4851911/e1aaccbf_K8sServicesTest_Checks_service_across_nodes_TFTP_with_DNS_Proxy_port_collision_Tests_TFTP_from_DNS_Proxy_Port.zip)

It happened in a backport PR.
https://github.com/cilium/cilium/pull/12332

**Comment by joestringer** [comment]
@jrajahalme This is a new test right? Would you be able to look into it?

**Comment by jrajahalme** [comment]
This is a different problem than what was fixed in #12248 (which was backported here).

Curl returns `72`:
```
CURLE_TFTP_UNKNOWNID (72)
    Unknown TFTP transfer ID.
```

In this test 10 curl commands are run from the same source port. According to logs the TFTP server is getting 10 requests, but apparently the 10th curl is receiving a TFTP transfer ID in reply that it did not send in the request. The Curl commands should run sequentially, so it should not be possible for the 10th curl to have received a duplicate of the 9th request?

This test does not need 10 rounds, as it is engineered to elicit a specific traffic pattern without any variation in the packet path. I'll change this test to do only one round, which hopefully resolves this flake.

**Comment by jrajahalme** [comment]
(hopefully) fixed by: #12462
